CMakeLists.txt

```diff
@@ -464,19 +489,40 @@ if(DEFINED CPPLINT_FILES)
     COMMAND ${CPPLINT_CMD} --quiet ${CPPLINT_ARGS} ${CPPLINT_FILES}
     WORKING_DIRECTORY ${P4C_SOURCE_DIR}
     COMMENT "cpplint quietly")
+endif()
+
+# clang-format
+# Temporarily disable clang-format linting for the ir folder.
+#TODO: Fix up the IR include system.
+list(FILTER P4C_LINT_LIST EXCLUDE REGEX "${P4C_SOURCE_DIR}/ir/")
+add_clang_format_files(${P4C_SOURCE_DIR} "${P4C_LINT_LIST}")
+
+find_program(CLANG_FORMAT_CMD clang-format)
+
+if(NOT ${CLANG_FORMAT_CMD})
+  if(DEFINED CLANG_FORMAT_FILES)
+    list(SORT CLANG_FORMAT_FILES)
+    set(CLANG_FORMAT_CMD clang-format)
+    add_custom_target(
+      clang-format
+      COMMAND ${CLANG_FORMAT_CMD} --Werror --dry-run -i ${CLANG_FORMAT_FILES}
```

Comment:
my version of clang-format does not understand these flags

Reply:
I would install the latest version (15.0.4) with `pip` and not with `apt`. I do not quite know how to bridge this otherwise.